scipy/signal/_fir_filter_design.py

```diff
@@ -834,6 +835,296 @@ def remez(numtaps, bands, desired, *, weight=None, type='bandpass',
+    N2 = (-20 * np.log10(np.sqrt(dp * ds)) - 13.0) / (14.6 * dF) + 1.0
+
+    return int(_oddceil(N2))
+
+
+def _remlplen_ichige(fp, fs, dp, ds):
+    """
+    Determine the length of the low pass filter with passband frequency
+    fp, stopband frequency fs, passband ripple dp, and stopband ripple ds.
+    fp and fs must be normalized with respect to the sampling frequency.
+    Note that the filter order is one less than the filter length.
+
+    Uses approximation algorithm described by Ichige et al.
+    """
+    dF = fs - fp
+
+    def v(dF, dp):
+        return 2.325 * ((-np.log10(dp)) ** -0.445) * dF ** (-1.39)
+
+    def arctan_g(fp, dF, d):
+        return (2.0 / np.pi) * np.arctan(
+            v(dF, dp) * (1.0 / fp - 1.0 / (0.5 - dF))
+        )
+
+    def arctan_h(fp, dF, c):
+        return (2.0 / np.pi) * np.arctan(
+            (c / dF) * (1.0 / fp - 1.0 / (0.5 - dF))
+        )
+
+    Nc = ceil(1.0 + (1.101 / dF) * (-np.log10(2.0 * dp)) ** 1.1)
+    Nm = (0.52 / dF) * np.log10(dp / ds) * (-np.log10(dp)) ** 0.17
+    N3 = ceil(
+        Nc
+        * (arctan_g(fp, dF, dp) + arctan_g(0.5 - dF - fp, dF, dp) + 1.0)
+        / 3.0
+    )
+    DN = ceil(
+        Nm
+        * (
+            arctan_h(fp, dF, 1.1)
+            - (arctan_h(0.5 - dF - fp, dF, 0.29) - 1.0) / 2.0
+        )
+    )
+    N4 = N3 + DN
+
+    return int(N4)
+
+
+def remezord(freqs, amps, rips, fs=1.0, alg="ichige"):
+    """
+    Filter parameter selection for the Remez exchange algorithm.
+
+    Calculate the parameters required by the Remez exchange algorithm to
+    construct a finite impulse response (FIR) filter that approximately meets
+    the specified design.
+
+    Parameters
+    ----------
+    freqs : array_like
+        A monotonic sequence of non-negative band edges in Hertz. All elements
+        must be less than half the sampling frequency ``fs``.
+    amps : array_like
+        A sequence containing the desired amplitudes of the signal to be
+        filtered over the corresponding frequency bands in ``freqs``.
+    rips : array_like
+        A sequence specifying the maximum allowable ripples of each band in the
+        frequency response.
+    fs : float, optional
+        The sampling frequency of the signal. Defaults to 1.0
+    alg : {'herrmann', 'kaiser', 'ichige'}, optional
+        Filter length approximation algorithm. Possible algorithms are
+        Herrmann [1]_, Kaiser [2]_, or Ichige [3]_. (Default: 'ichige')
+
+    Returns
+    -------
+    numtaps : int
+        The desired number of taps in the FIR filter.
+    bands : ndarray
+        A monotonic sequence containing the band edges.
+    desired : ndarray
+        A sequence half the size of bands containing the desired gain in each
+        of the specified bands.
+    weight : ndarray
+        A relative weighting to give to each band region.
+
+    Raises
+    ------
+    ValueError:
+        - If any element in ``freqs`` is negative, greater than 0.5, or the
```

Comment:
Parameters should be in single backticks
```suggestion
        - If any element in `freqs` is negative, greater than 0.5, or the
```